app/charts/map/chart-map-prototype.tsx

```diff
@@ -65,74 +55,78 @@ export const ChartMapVisualization = ({
   const [geoData, setGeoData] = useState<GeoDataState>({ state: "fetching" });
   const locale = useLocale();
 
-  const areaLayerComponentIri = chartConfig.fields.areaLayer.componentIri;
-  const symbolLayerComponentIri = chartConfig.fields.symbolLayer.componentIri;
+  const areaGeoDimensionIri = chartConfig.fields.areaLayer.componentIri;
+  const symbolGeoDimensionIri = chartConfig.fields.symbolLayer.componentIri;
 
   const [{ data, fetching, error }] = useDataCubeObservationsQuery({
     variables: {
       locale,
       iri: dataSetIri,
-      measures: [areaLayerComponentIri, symbolLayerComponentIri],
+      measures: [areaGeoDimensionIri, symbolGeoDimensionIri],
       filters: queryFilters,
     },
   });
   const dimensions = data?.dataCubeByIri?.dimensions;
+  const measures = data?.dataCubeByIri?.measures;
   const observations = data?.dataCubeByIri?.observations.data;
 
-  const areaLayerDimension = dimensions?.find(
-    (d) => d.iri === areaLayerComponentIri
-  ) as GeoShapesDimension | undefined;
-
   const areaLayer = useMemo(() => {
-    if (areaLayerDimension) {
-      const rawShapes = areaLayerDimension.geoShapes as GeoShapes;
-      const topojsonShapes = topojsonFeature(
-        rawShapes,
-        rawShapes.objects.shapes
-      ) as GeoJSON.FeatureCollection<
-        GeoJSON.GeometryObject,
-        ShapeFeature["properties"]
-      >;
-
-      topojsonShapes.features.forEach((d) => {
-        // Should we match by labels?
-        d.properties.observation = observations?.find(
-          (o) => o[areaLayerComponentIri] === d.properties.label
-        );
-      });
-
-      return {
-        shapes: topojsonShapes,
-        mesh: topojsonMesh(rawShapes, rawShapes.objects.shapes as any),
-      };
+    if (dimensions) {
```

Comment:
Nit: to prevent a big diff while only the indentation level has changed, I think the early return pattern can be used.

if (!dimensions) {
  return
}